File: Semana 2/LastDigitoftheSumofSquaresofFibonacciNumbers.py
def SumSquaresFibonacciNumbers(number):
    if number < 2:
        return number
    aux = 0
    aux1 = 0
    result = 1
    for i in range(1,number):
        aux = result
        result = result+aux1
        aux1 = aux
    return (result + aux1) * result
if __name__ == "__main__":
    number = int (input())
    print(SumSquaresFibonacciNumbers(number)%10)

File: Semana 2/LastDigitoftheSumofSquaresofFibonacciNumbers.cpp
#include <iostream>
using namespace std;

long long SumSquaresFibonacciNumbers(long long n){

    if(n < 2){
        return n;
    }
    long long aux1 = 0;
    long long aux2 = 1;
    long long numero = 1;
    for (long long i  = 0; i < n-1 ; i ++){
        aux2 = numero;
        numero += aux1;
        aux1 = aux2;
    }
    return ((numero + aux1) * numero)%10;
}

int main (){
    long long number;
    cin >> number;
    cout << SumSquaresFibonacciNumbers(number) << endl;
}